Main menu caesars cipher

Vladimir2075 opened the pull request.

Vladimir2075 committed:
implemented MainMenu

Vladimir2075 committed:
implemented drawing menu

Vladimir2075 committed:
1. implemented navigation on menu
2. Creating a class CaesarsCipher using a pattern singleton

Vladimir2075 committed:
1.Implement path validation
2. Implementation of encryption and decryption by method Caesars

Volodymyr95 commented on src/CaesarsCipher.java:
```diff
@@ -0,0 +1,106 @@
+
+    private CaesarsCipher() {
+    }
+    public  static synchronized CaesarsCipher  getInstance() {
```
don't use synchronized

Volodymyr95 commented on src/CaesarsCipher.java:
```diff
@@ -0,0 +1,106 @@
+        int  currPosition =  CYR_LETTERS.indexOf(ch);
+        int newPosition;
+        if (currPosition ==-1) {
+            return ch;
```
ch is not informative

Volodymyr95 commented on src/CaesarsCipher.java:
```diff
@@ -0,0 +1,106 @@
+    private char newChar (char ch, int key) {
+        int  currPosition =  CYR_LETTERS.indexOf(ch);
+        int newPosition;
+        if (currPosition ==-1) {
```
-1 is magic number

Volodymyr95 commented on src/CaesarsCipher.java:
```diff
@@ -0,0 +1,106 @@
+    }
+
+    void encryptionDecryption (boolean isEncryption){
+        int localKey =  isEncryption?KEY:(-1*KEY);
```
isEncryption ? KEY : ( -1 * KEY)

Volodymyr95 commented on src/MainMenu.java:
```diff
@@ -0,0 +1,66 @@
+        RunMainMenu(0);
+    }
+
+    //level =1  - головне меню; level =2- підменю
```
delete comment from here

Volodymyr95 commented on src/MainMenu.java:
```diff
@@ -0,0 +1,66 @@
+            if (keyboard.hasNextInt()) {
+                yourChoice = keyboard.nextInt();
+            }
+            if (level == 0) {
```
0,1,2,3 are magic numbers

Volodymyr95 commented on src/MainMenu.java:
```diff
@@ -0,0 +1,66 @@
+        }
+    }
+
+    private void menuDrawing (String arg1, String arg2, String arg3){
```
arg1, arg2,arg3 are not informative

Vladimir2075 committed:
Fixed of all remarks:
1.don't use synchronized in src/CaesarsCipher.java
2.-1 is magic number in src/CaesarsCipher.java ->private char newChar
3.isEncryption ? KEY : ( -1 * KEY)
4.delete comment  //level =1  - головне меню; level =2- підменю
5. 0,1,2,3 are magic numbers in src/MainMenu.java
6. arg1, arg2,arg3 are not informative

Volodymyr95 commented on src/CaesarsCipher.java:
```diff
@@ -0,0 +1,108 @@
+        return resultPath;
+    }
+
+    void encryptionDecryption (boolean isEncryption){
```
why default access modifier?

Volodymyr95 commented on src/MainMenu.java:
```diff
@@ -0,0 +1,71 @@
+    private final CaesarsCipher CAESARS_CIPHER=CaesarsCipher.getInstance();
+    public MainMenu() {
+        System.out.println("Головне меню:");
+        RunMainMenu(0);
```
why 0?

Volodymyr95 commented on src/MainMenu.java:
```diff
@@ -0,0 +1,71 @@
+    final String NAME_OPTION_EXIT = "3.Вихід (3)";
+
+    final int MAIN_LEVEL=0;
+    final int MAIN_SECOND_LEVEL=1;
```
final int MAIN_SECOND_LEVEL = 1; fix formattting for all files

Vladimir2075 committed:
fixed:
1. why default access modifier?
2. final int MAIN_SECOND_LEVEL = 1; fix formattting for all files
3. why 0?

Volodymyr95 commented on src/CaesarsCipher.java:
```diff
@@ -0,0 +1,108 @@
+import java.util.Scanner;
+
+public class CaesarsCipher {
+    private static CaesarsCipher instance;
```
make it final

Volodymyr95 commented on src/CaesarsCipher.java:
```diff
@@ -0,0 +1,108 @@
+        return instance;
+    }
+
+    private char newChar (char currentChar, int key) {
```
not informative name

Volodymyr95 commented on src/MainMenu.java:
```diff
@@ -0,0 +1,74 @@
+    public void RunMainMenu (int level){
+        boolean isExit = false;
+        Scanner keyboard = new Scanner(System.in);
+        if (level ==MAIN_FIRST_LEVEL) {
```
level == MAIN_FIRST_LEVEL - formatting

Volodymyr95 commented on src/CaesarsCipher.java:
```diff
@@ -0,0 +1,108 @@
+    }
+
+    private char newChar (char currentChar, int key) {
+        int  currPosition =  CYR_LETTERS.indexOf(currentChar);
```
currentPosition
